### DISCUSSION TITLE
ZIP only files

### DISCUSSION THREAD

**Discussion by ivan200** [rejetto/hfs#640] [Q&A]
On my server, I have a large number of files organized in a tree-like structure. The total size of these files is approximately 100 GB, (200,000 files and 10,000 folders containing files of 10-300 KB each). 
Sometimes, some users select one or more of the basic folders and attempt to download them (like try to download whole my server without knowing how big it is)
Currently, I have blocked the zipping process entirely. However, I would still like to allow for users for the downloading of 1-100 selected files.
It would be great for me if one of these functions were there.:

- Prevent folders from being zipped, but allow for the selection and zip of individual files. 

- or somehow limit the zipping of the selection based on its size. For example, allow zipping only if less than 100 MB of files and folders in total are selected (or one file greater)

- or limit the zipping of folders if there is at least one folder inside

is it possible?

**Comment by rejetto** [comment]
i will investigate first option in 0.54

**Comment by rejetto** [comment]
Ok, you can achieve this behavior (the first you described).
1. Go to Admin-panel > Share files
2. select the home folder
3. set "Who can zip" to "no one"
4. set this text in the "Masks" field
```
"**":
  maskOnly: files
  can_archive: true
```

Doing so, if the user selects a file and a folder, the zip will contain only the file.
This is not ideal, as one may think the folder was included, but still the result is the correct one.
To avoid this kind of confusion, in next version (not yet published) I'm trying to prevent people from selecting items that cannot be zipped or deleted, as the selection is useful only for these actions.

**Reply by ivan200** [reply]
ty very much
